MudTabPanel: Add ShowCloseIcon property

Yomodo opened the pull request:
## Description
This PR adds the `ShowCloseIcon=true` property to `MudTabPanel` to show/hide the close icon
when using `MudDynamicTabs`.

Use cases:
-Sometimes we just need to prevent closing a tab, e.g. stuff still in progress;
-Sometimes a minimal tab of 1 is mandatory;

![image](https://user-images.githubusercontent.com/10358198/210024755-99ac4e62-ecca-4860-919c-40771a29e4a6.png)

![image](https://user-images.githubusercontent.com/10358198/210024696-7a98c106-1c87-495e-a228-73ed303a2f73.png)



## How Has This Been Tested?
<!-- All PR's should implement unit tests if possible -->
<!-- Please describe how you tested your changes. -->
<!-- Have you created new tests or updated existing ones? -->
unit | visually

## Types of changes
-Updated docs and some doc corrections.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

<!-- If you made any visual changes, provide screenshots of before/after, it its moving parts, please provide high quality gif, wemb or mp4 -->

## Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] The PR is submitted to the correct branch (`dev`).
- [X] My code follows the code style of this project.
- [X] I've added relevant tests.

Yomodo committed:
Added ShowCloseIcon property, updated docs

Yomodo committed:
Added ShowCloseIcon property, updated docs

Yomodo committed:
Fixed formatting

Yomodo committed:
Formatting

Yomodo committed:
Formatting

Yomodo committed:
Revert "Formatting"

This reverts commit 904559f6dd277e522c2da425222c5f290009b395.

Yomodo committed:
Revert "Formatting"

This reverts commit 904559f6dd277e522c2da425222c5f290009b395.

iwis commented on src/MudBlazor/Components/Tabs/MudTabPanel.razor:
```diff
@@ -46,6 +46,13 @@ else
+    /// MudDynamicTabs: If true, shows the close icon in the TabPanel.
+    /// </summary>
+    [Parameter]
+    [Category(CategoryTypes.Tabs.Appearance)]
```
That isn't the Appearance category but the Behavior category, because it changes not only the appearance of an application but also its behavior :) I described the difference between both categories here: https://github.com/MudBlazor/MudBlazor/blob/dev/src/MudBlazor/Attributes/CategoryAttribute.cs#L92

Yomodo replied:
Right, sorry you had to repeat yourself :)

Yomodo committed:
Corrected property CategoryTypes

Yomodo committed:
Corrected property CategoryTypes

mikes-gh commented:
Thanks @Yomodo 
I know its frustrating but could you fix the indenting.
Looks like the previous developer used Tabs for indents instead of spaces.

Yomodo commented:
> I know its frustrating but could you fix the indenting.
> Looks like the previous developer used Tabs for indents instead of spaces.

Indeed, but surely it is acceptable since indents are now spaces and the formatting overall still looks ok?
I did try to correct indenting but only made it worse and reverted that,
I really don't want to spent more time on that. I tried.

Yomodo commented:
I'll try again.

Yomodo committed:
indent fixing, another attempt

Yomodo committed:
indent fixing, another attempt

Yomodo committed:
MudTable: Fix 2nd level checkbox default indeterminate state (#6075)

Yomodo committed:
Docs: added note about Elevation and Outlined properties (#6071)

henon committed:
Docs: Show how to use custom SVG icons

BieleckiLtd committed:
Error box: Change background when circuit (connection) is lost. (#6115, #6116)

mckaragoz committed:
MudDivider: Fix fullwidth behavior in flex containers (#5458, #5472)

rogue6800 committed:
MudTooltip: Fix Style not forwarded to popover (#6045, #6051)

Suenodk committed:
MudDialog: Added Draggable as a DialogOption. (#6083)

henon committed:
Revert "MudDialog: Added Draggable as a DialogOption. (#6083)" due to reported dragging issues.

This reverts commit 52134a580020f397acd26916667852fa79893502.

dependabot[bot] committed:
Build(deps): Bump bunit from 1.13.5 to 1.14.4 in /src (#6156)

Bumps [bunit](https://github.com/bUnit-dev/bUnit) from 1.13.5 to 1.14.4.
- [Release notes](https://github.com/bUnit-dev/bUnit/releases)
- [Changelog](https://github.com/bUnit-dev/bUnit/blob/main/CHANGELOG.md)
- [Commits](https://github.com/bUnit-dev/bUnit/compare/v1.13.5...v1.14.4)

---
updated-dependencies:
- dependency-name: bunit
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump Microsoft.AspNetCore.Components.WebAssembly in /src (#6159)

Bumps [Microsoft.AspNetCore.Components.WebAssembly](https://github.com/dotnet/aspnetcore) from 7.0.1 to 7.0.2.
- [Release notes](https://github.com/dotnet/aspnetcore/releases)
- [Changelog](https://github.com/dotnet/aspnetcore/blob/main/docs/ReleasePlanning.md)
- [Commits](https://github.com/dotnet/aspnetcore/compare/v7.0.1...v7.0.2)

---
updated-dependencies:
- dependency-name: Microsoft.AspNetCore.Components.WebAssembly
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump Microsoft.AspNetCore.Components.WebAssembly.DevServer (#6158)

Bumps [Microsoft.AspNetCore.Components.WebAssembly.DevServer](https://github.com/dotnet/aspnetcore) from 7.0.1 to 7.0.2.
- [Release notes](https://github.com/dotnet/aspnetcore/releases)
- [Changelog](https://github.com/dotnet/aspnetcore/blob/main/docs/ReleasePlanning.md)
- [Commits](https://github.com/dotnet/aspnetcore/compare/v7.0.1...v7.0.2)

---
updated-dependencies:
- dependency-name: Microsoft.AspNetCore.Components.WebAssembly.DevServer
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Build(deps): Bump Microsoft.AspNetCore.Components.WebAssembly.Server (#6157)

Bumps [Microsoft.AspNetCore.Components.WebAssembly.Server](https://github.com/dotnet/aspnetcore) from 7.0.1 to 7.0.2.
- [Release notes](https://github.com/dotnet/aspnetcore/releases)
- [Changelog](https://github.com/dotnet/aspnetcore/blob/main/docs/ReleasePlanning.md)
- [Commits](https://github.com/dotnet/aspnetcore/compare/v7.0.1...v7.0.2)

---
updated-dependencies:
- dependency-name: Microsoft.AspNetCore.Components.WebAssembly.Server
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Mr-Technician committed:
MudTable: Add null check on server data items in FilteredItems (#6186)

Mr-Technician committed:
MudTable: Enable FluentValidation for TableRowValidator. (#6193)

Yomodo committed:
MudTabs: Fix incorrect ActivePanelIndex (#6100)

mikes-gh commented:
Hi @Yomodo could you fix the conflicts please Thanks

Yomodo commented:
Will do!

Yomodo committed:
rebase

Yomodo commented:
Now sure how, but it got messy :(
I will close this PR and open a new one.
